### DISCUSSION TITLE
Return isExecuting and isIdle in useAction

### DISCUSSION THREAD

**Discussion by anarkrypto** [next-safe-action/next-safe-action#129] [Ideas]
Example:
```js
const { isExecuting, isIdle } = useAction(...)
```

I find it much more useful to check with `isExecuting` than doing ```status === 'executing'``` checks throughout the code.

I don't see a reason to return only the status as a string. Popular libs like **react-query** and **SWR** has `isLoading`

**Comment by TheEdoRan** [comment]
Check out [this page](https://next.next-safe-action.dev/docs/execution/hooks/check-action-status#utility-functions) in docs.

**Reply by anarkrypto** [reply]
Got it. However this solutions does not allow me to track more than 1 action individually.

**Reply by TheEdoRan** [reply]
You can rename the `status` property when destructuring the hook object:

```tsx
const { status: action1Status } = useAction(action1);
const { status: action2Status } = useAction(action2);
```

You'd have to do the same thing with an `isExecuting` property, because of the name conflict.

**Reply by anarkrypto** [reply]
Yes, but it's only possible to rename when the status comes from useAction. I mean, it makes no difference rename when importing from next-safe-action/status.

**Reply by TheEdoRan** [reply]
That's a valid point. I mean, in your components you could write: 

```tsx
const isAction1Executing = isExecuting(action1status);
const isAction2Executing = isExecuting(action2status);
```

But it's definitely annoying to do that just to check action execution status.

I could keep `status` property to allow for discriminated checks and re-add `isIdle`, `isExecuting`, `hasSucceeded` and `hasErrored` properties in hook return objects (as pre v4). I think it's a good compromise. What do you think?

**Reply by TheEdoRan** [reply]
Shorthand statuses are now returned from hooks in [v7.0.0-next.41](https://github.com/TheEdoRan/next-safe-action/releases/tag/v7.0.0-next.41)

**Reply by anarkrypto** [reply]
That's awesome! Thank you!
